### DISCUSSION TITLE
Code Blocks (PDF): option to configure style and other things for float type `codelisting`

### DISCUSSION THREAD

**Discussion by joelvonrotz** [quarto-dev/quarto-cli#7477] [Q&A] [answered]
### Description

Hello

I'm creating a template for my upcoming industry thesis & bachelor thesis. I've noticed, that the style of the float type `codelisting`, which is used for code block captions, is not really by my liking. My current workaround is following:

Inside a config file, which is attached via `include-in-header`, I define my new float style:

```latex
\usepackage{floatrow}
\DeclareFloatStyle{MyListingStyle}
  {
    style=plaintop,
    captionskip=1pt
  }
```

In the main Quarto-document right at the beginning of it, I update the style to my own one:

```latex
% beginning of main qmd-document
\floatsetup[codelisting]{style=MyListingStyle}

% other content
```

Is there an easier way to do this?

**Comment by cscheid** [comment, marked as answer]
That's most likely the easiest way. Because you're making changes to the LaTeX rendering, you'll need to, eventually somewhere, make LaTeX changes.

**Reply by mcanouil** [reply]
(Changed the thread to Q&A)

**Reply by joelvonrotz** [reply]
Ah, thanks for your response! Then I'll go with this, thanks!
